### PR TITLE
[PlacementGroup]Add guarded by in placement group scheduler ut

### PR DESCRIPTION
**Pull request by clay4megtr**
## Why are these changes needed?
add  GUARDED_BY annotation for gcs scheduler ut.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:71]:
```suggestion
  std::vector<std::shared_ptr<gcs::GcsPlacementGroup>> &GetSuccessPlacementGroups() {
```

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:76]:
```suggestion
  std::vector<std::shared_ptr<gcs::GcsPlacementGroup>> &GetFailurePlacementGroups() {
```

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: updated.

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: udpated.

**Comment by rkooo567** [comment]
@clay4444. Can you assign me to the "assignee" section instead? That's the formal process we are trying to use.

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (rkooo567) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:71]: This is not really safe right? This vector still can be modified?

**Review by rkooo567 — COMMENTED** [review]
Can you add `GUARDED_BY` to all vector attributes?

**Comment by clay4444** [comment]
> @clay4444. Can you assign me to the "assignee" section instead? That's the formal process we are trying to use.

yes, I get it~ :)

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: yes, I have delete it,  I have made a specialized method for ut check where this two vector used, maybe you can review this  :)

**Comment by clay4444** [comment]
> Can you add `GUARDED_BY` to all vector attributes?

hi, @rkooo567  I don't quite understand the  'all vector attributes',  do you mean all vector need add `GUARDED_BY ` ?   in the code,  only `success_placement_groups_` and `failure_placement_groups_` will modified by multi thread.

**Comment by rkooo567** [comment]
@clay4444 Yes! I didn't know the first vector wasn't used by other threads (I just assumed this lock should guard all the vector attributes based on the name "vector_lock"). The other vector is `raylet_clients_`.

**Comment by clay4444** [comment]
> @clay4444 Yes! I didn't know the first vector wasn't used by other threads (I just assumed this lock should guard all the vector attributes based on the name "vector_lock"). The other vector is `raylet_clients_`.

yes, the name "vector_lock" is confused indeed, do you think we need change it to "placement_groups_requests_lock_" or others ?

**Comment by rkooo567** [comment]
SGTM :)!

**Comment by clay4444** [comment]
> SGTM :)!

yes~ I have change it to "placement_group_requests_mutex_"  :)

**Comment by rkooo567** [comment]
Please remove the `@author-action-required` tag once it is ready to go :)

**Comment by clay4444** [comment]
> Please remove the `@author-action-required` tag once it is ready to go :)

ok~ 😄

**Comment by clay4444** [comment]
@ffbin @rkooo567 , is this can be merge ?

**Comment by rkooo567** [comment]
I’ll review it again by today

**Comment by clay4444** [comment]
> I’ll review it again by today

yeah~  thank u  :)

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:59]:
```suggestion
                                     const GcsPlacementGroupStatus &status) {
```

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:65]: return status == GcsPlacementGroupStatus::SUCCESS ? success_placement_groups_.size() == expected_count : failure_placement_groups_.size() == expected_count;

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:71]:
```suggestion
  void CheckPlacementGroupSize(int expected_count, const GcsPlacementGroupStatus &status) {
```

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:82]:
```suggestion
      const GcsPlacementGroupStatus &status) {
```

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (ffbin) [inline src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:58]:
```suggestion
  void WaitPlacementGroupPendingDone(int expected_count,
```

Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc:58]: the reason use  unsigned int is `GcsPlacementGroupStatus` enum is `uint32_t`,  I will change it to `int32_t`

**Review by rkooo567 — APPROVED** [review]
I think it is good to go if you address @ffbin's comments. Please remove the tag once you are ready! I will merge them :)

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: updated

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: updated

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: updated

File: src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc
Comment (clay4444) [reply src/ray/gcs/gcs_server/test/gcs_placement_group_scheduler_test.cc]: updated

**Comment by rkooo567** [comment]
Please add the `test_ok` tag if tests look good to go!

**Comment by clay4444** [comment]
> Please add the `test_ok` tag if tests look good to go!

ok~

**Review by ffbin — APPROVED** [review]

**Comment by clay4444** [comment]
hi, @rkooo567 , I always encounter timeout problems with python ut, the code segment is `_wait_for_node` method in cluster_utils.py,  I think this is not related to the current pr, I have tried hard to fix it, but it will hardly reappear on my machine.....   cc @ffbin

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**
